manifests/server.pp

```diff
@@ -27,16 +28,25 @@
 
   if $use_xinetd {
     include xinetd
+    $ensure = $enable ? {
+      true  => 'present',
+      false => 'absent'
+    }
     xinetd::service { 'rsync':
+      ensure      => $ensure,
       bind        => $address,
       port        => '873',
       server      => '/usr/bin/rsync',
       server_args => "--daemon --config ${conf_file}",
       require     => Package['rsync'],
     }
   } else {
+    $ensure = $enable ? {
+      true  => 'running',
+      false => 'stopped'
+    }
     service { $servicename:
-      ensure     => running,
+      ensure     => $ensure,
       enable     => true,
```

Comment:
If the `service_ensure => 'stopped'` then `enable` should probably be `false` so it doesn't start at boot when not using xinetd.